### PR TITLE
Add `contributing.md`, and clean up some other docs.

### PR DESCRIPTION
**Pull request by rfk**
I'm trying to iterate on the contributor experience for UniFFI,
both to support new team members coming on board and to support
additional external contributors. As a start, this:

* Adds a contributing.md file with some concrete steps on how
  to get up and running.
* Cleans up the README, linking out to more thorough docs
  rather than brain-dumping a bunch of stuff into a single place.
  * In particular, some motivational material has moved into a
    new "Motivation" section of the user guide.
* Moves all scaffolding-related code into its own sub-module,
  to make it easier to describe how to navigate the code.

For convenience, here is the **[Rendered View of the new contributing document](https://github.com/mozilla/uniffi-rs/blob/contributing-dot-md/docs/contributing.md)**.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -90,116 +66,4 @@ As of version 0.11.0, all interfaces will be required to be  `Send+Sync`, and th
 `[Threadsafe]` attribute will be deprecated and ignored.
 
 See also [adr-0004](https://github.com/mozilla/uniffi-rs/blob/main/docs/adr/0004-only-threadsafe-interfaces.md)
-which outlines the reasoning behind this decision.
-
-### Component Interface
```
Comment (rfk) [inline README.md:95]: These details on "how does UniFFI work" are mostly duplicated in the user guide, so I deleted them from here.

File: uniffi_bindgen/askama.toml [hunk 2]
```diff
@@ -1,6 +1,6 @@
 [general]
 # Directories to search for templates, relative to the crate root.
-dirs = [ "src/templates", "src/bindings/kotlin/templates", "src/bindings/python/templates", "src/bindings/swift/templates", "src/bindings/gecko_js/templates", "src/bindings/ruby/templates" ]
+dirs = [ "src/scaffolding/templates", "src/bindings/kotlin/templates", "src/bindings/python/templates", "src/bindings/swift/templates", "src/bindings/gecko_js/templates", "src/bindings/ruby/templates" ]
```
Comment (rfk) [inline uniffi_bindgen/askama.toml:3]: Moving this into a subdirectory was entirely motivated by making it easier to write the "Navigating the code" section of the contributor doc.

File: docs/contributing.md [hunk 3]
```diff
@@ -0,0 +1,134 @@
...
+  * `swift` and `swiftc`, the [Swift command-line tools](https://swift.org/download/).
+  * The Swift `Foundation` package.
+* Python:
+  * A `python3` interpreter.
+* Ruby:
+  * A `ruby` interpreter.
+  * The [`FFI`](https://github.com/ffi/ffi) Ruby gem, installable via `gem install FFI`.
+
+
+## Navigating the code
```
Comment (skhamis) [inline docs/contributing.md:63]: Very nice section! This will definitely make it easier even when coming back to uniffi after a brief hiatus!

**Review by skhamis — APPROVED** [review]
From my brief on boarding experience I think this improves the documentation on a bunch of layers (technical, contributing, etc). Anything I was confused or unsure about is pretty clear now (thanks to contributing.md and navigating the code sections)!

**Pull request merged**